### PR TITLE
chore(buildtool): implementation cleanup

### PR DESCRIPTION
**Pull request by ewiseblatt**
Changed strategy for dealing with nebula friction by using a decoy origin
repository rather than removing all the confusing tags and versions.
Using the nebula release.verison switch eliminates the need to remove tags,
however it insists on pushing a new tag (and the wrong pattern at that).
The decoy repository tricks nebula into pushing the tag to somewhere irrelvant.

Cleaned up some of the gradle build command classes, adding build_rpms

@jtk54 
This contains the other changes I was working on before fixing nebula.
I combined them together because it was easier for me and doesnt affect anyone else yet.

### CODE REVIEW

File: dev/buildtool/build_commands.py [hunk 1]
```diff
@@ -32,15 +33,101 @@
...
+
+    summary_match = re.match(
+        r"[> ]*(.*?)'(.+?)': HTTP/[0-9\.]+ ([0-9]{3}) ([\w]+)(.*)",
+        summary_line)
+    failed_by = 'unknown'
+    failed_reason = 'unknown'
+    if not summary_match:
+      logging.error('Cannot interpret cause of failure:\n%s', summary_line)
+    else:
+      # abstract = summary_match.group(1)
```
Comment (jtk54) [inline dev/buildtool/build_commands.py]: Will be used in the future?

File: dev/buildtool/build_commands.py [hunk 2]
```diff
@@ -234,272 +306,127 @@ def __is_newer_repo(self, test_path, ref_path, log):
...
-        options.scratch_dir, name, '{name}-summary.yml'.format(name=name))
-    write_to_path(summary.to_yaml(with_commit_messages=False), summary_path)
-    return nebula_repo_path
+    jar_repo = options.build_jar_repository
+    if not jar_repo or len(jar_repo.split('/')) != 1:
+      raise ValueError('Expected --build_jar_repository in the form <repo>'
+                       ' using the implied owner from'
+                       ' --build_bintray_repository')
+
+    parts = (options.build_bintray_repository or '').split('/')
```
Comment (jtk54) [inline dev/buildtool/build_commands.py]: It always seemed weird that we were explicitly passing a single jar repo string, but the org and deb package repo as `{org}/{deb_repo}`.

Comment (ewiseblatt) [reply dev/buildtool/build_commands.py]: I changed the config options. have a look.

File: dev/buildtool/build_commands.py (hunk 1 above)
Comment (ewiseblatt) [reply dev/buildtool/build_commands.py]: i dont know. it's there for documentation. As I encounter and handle more error cases I might change the label schema to better accomodate them. For now we just want to be able to characterize and quantify impediments to building artifacts.

**Pull request merged**